packages/appstore/src/components/main-title-bar/index.tsx

```diff
@@ -16,12 +15,12 @@ const WalletsBanner = makeLazyLoader(
 )();
 
 const MainTitleBar = () => {
-    const { is_wallet_enabled } = useFeatureFlags();
     const { traders_hub, client } = useStore();
     const { selected_region, handleTabItemClick, toggleRegulatorsCompareModal, content_flag } = traders_hub;
-    const { is_landing_company_loaded, is_switching } = client;
+    const { is_landing_company_loaded, is_switching, wallet_migration_state } = client;
     const is_low_risk_cr_real_account =
         content_flag === ContentFlag.LOW_RISK_CR_NON_EU || content_flag === ContentFlag.LOW_RISK_CR_EU;
+    const show_wallets_banner = wallet_migration_state && wallet_migration_state !== 'ineligible';
```

Comment:
@sergei-deriv there is a check for the wallet's state in the component itself and it returns null if the state is anything rather than eligible, failed and in_progress, why do we need the state here?

Reply:
@mahdiyeh-deriv we need to check the state here to define we need to download wallet banners or not. In other cases we will download all wallet images (it's about 0.5MB) for all users (for eligable and ineligable for migrate to wallets too), doesn't matter we need to show it or not.